scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java

```diff
@@ -0,0 +1,436 @@
+
+import java.security.KeyException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+
+/**
+ *
+ */
+public class SchedulerNodeClient implements ISchedulerClient {
+
+    private static final Logger logger = Logger.getLogger(SchedulerNodeClient.class);
+
+    private Decrypter decrypter;
+
+    private String url;
+
+    private ISchedulerClient client;
+
+    public SchedulerNodeClient(Decrypter decrypter, String url) {
+        SchedulerNodeClient.this.decrypter = decrypter;
+        SchedulerNodeClient.this.url = url;
+        client = SchedulerClient.createInstance();
+
+    }
+
+    public void connect() throws Exception {
+        connect(SchedulerNodeClient.this.url);
+    }
+
+    public void connect(String url) throws Exception {
+        System.out.println("[SchedulerNodeClient] Connecting to scheduler at url " + url);
+        CredData userCreds = decrypter.decrypt();
+        client.init(new ConnectionInfo(url, userCreds.getLogin(), userCreds.getPassword(), null, true));
+        System.out.println("[SchedulerNodeClient] Connected");
+    }
+
+    public List<JobUsage> getMyAccountUsage(Date startDate, Date endDate) throws NotConnectedException, PermissionException {
+        return client.getMyAccountUsage(startDate, endDate);
+    }
+
+    public List<JobUsage> getAccountUsage(String user, Date startDate, Date endDate) throws NotConnectedException, PermissionException {
+        return client.getAccountUsage(user, startDate, endDate);
+    }
+
+    public List<String> getUserSpaceURIs() throws NotConnectedException, PermissionException {
+        return client.getUserSpaceURIs();
+    }
+
+    public List<String> getGlobalSpaceURIs() throws NotConnectedException, PermissionException {
+        return client.getGlobalSpaceURIs();
+    }
+
+    public JobResult getJobResult(JobId jobId) throws NotConnectedException, PermissionException, UnknownJobException {
+        return client.getJobResult(jobId);
+    }
+
+    public TaskResult getTaskResultFromIncarnation(JobId jobId, String taskName, int inc) throws NotConnectedException, UnknownJobException, UnknownTaskException, PermissionException {
+        return client.getTaskResultFromIncarnation(jobId, taskName, inc);
+    }
+
+    public boolean killTask(JobId jobId, String taskName) throws NotConnectedException, UnknownJobException, UnknownTaskException, PermissionException {
+        return client.killTask(jobId, taskName);
+    }
+
+    public boolean restartTask(JobId jobId, String taskName, int restartDelay) throws NotConnectedException, UnknownJobException, UnknownTaskException, PermissionException {
+        return client.restartTask(jobId, taskName, restartDelay);
+    }
+
+    public boolean preemptTask(JobId jobId, String taskName, int restartDelay) throws NotConnectedException, UnknownJobException, UnknownTaskException, PermissionException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)